lib/ProductOpener/Tags.pm

```diff
@@ -1702,6 +1704,188 @@ sub build_tags_taxonomy($$$) {
+				if ((not defined $fields_ref) or (defined $fields_ref->{children})) {
+					exists $taxonomy_ref->{$tagid}{children} or $taxonomy_ref->{$tagid}{children} = [];
+					push @{$taxonomy_ref->{$tagid}{children}}, $childid;
+				}
+				if (($options_ref->{include_children}) and ($requested_tags{$tagid})) {
+					push @tags, $childid;
+				}
+			}
+		}
+
+		if (((not defined $fields_ref) or (defined $fields_ref->{name}))
+			and (defined $translations_to{$tagtype}{$tagid})) {
+
+			$taxonomy_ref->{$tagid} = {name => {}};
+
+			foreach my $lc (@{$lcs_ref}) {
+
+				if (defined $translations_to{$tagtype}{$tagid}{$lc}) {
+					$taxonomy_ref->{$tagid}{name}{$lc} = $translations_to{$tagtype}{$tagid}{$lc};
+				}
+
+				my $lc_tagid = get_string_id_for_lang($lc, $translations_to{$tagtype}{$tagid}{$lc});
+
+				if (defined $synonyms_for{$tagtype}{$lc}{$lc_tagid}) {
+
+					# additives has e-number as their name, and the first synonym is the additive name
+					if (($tagtype =~ /^additives(|_prev|_next|_debug)$/) and (defined $synonyms_for{$tagtype}{$lc}{$lc_tagid}[1])) {
+						$taxonomy_ref->{$tagid}{name}{$lc} .= " - " . $synonyms_for{$tagtype}{$lc}{$lc_tagid}[1];
+					}
+
+					# add synonyms to the full taxonomy
+					if (((not defined $fields_ref) or (defined $fields_ref->{synonyms}))
+						and (defined $synonyms_for{$tagtype}{$lc}{$lc_tagid})) {
+						(defined $taxonomy_ref->{$tagid}{synonyms}) or $taxonomy_ref->{$tagid}{synonyms} = {};
+						$taxonomy_ref->{$tagid}{synonyms}{$lc} = $synonyms_for{$tagtype}{$lc}{$lc_tagid};
+					}
+				}
+			}
+		}
+
+		if (defined $properties{$tagtype}{$tagid}) {
+
+			foreach my $prop_lc (keys %{$properties{$tagtype}{$tagid}}) {
+				
+				if ($prop_lc =~ /^(.*):(\w\w)$/) {
+					my $prop = $1;
+					my $lc = $2;
+
+					if (
+						# Include the property in all requested languages if the property
+						# is specified without a language in the fields parameter
+						# or if the fields parameter is not specified.
+						
+						(((not defined $fields_ref) or (defined $fields_ref->{$prop}))
+							and (grep( /^$lc$/, @$lcs_ref)))
+
+						# Also include the property if it was requested in a specific language
+						# e.g. fields=vegan:en
+						# as some properties are defined only for English
+
+						or ((defined $fields_ref) and (defined $fields_ref->{$prop_lc}))
+						
+						) {
+
+						(defined $taxonomy_ref->{$tagid}{$prop}) or $taxonomy_ref->{$tagid}{$prop} = {};
+						$taxonomy_ref->{$tagid}{$prop}{$lc} = $properties{$tagtype}{$tagid}{$prop_lc};
+					}
+				}
+			}
+		}
+
+		# Allow requests for inherited properties
+		foreach my $property_ref (@properties) {
+
+			my $prop = $property_ref->[0];
+			my $lc = $property_ref->[1];
+			my $property_value = get_inherited_property($tagtype, $tagid, "$prop:$lc");
+			if (defined $property_value) {
+				(defined $taxonomy_ref->{$tagid}{$prop}) or $taxonomy_ref->{$tagid}{$prop} = {};
+					$taxonomy_ref->{$tagid}{$prop}{$lc} = $property_value;
```

Comment:
```suggestion
                    # we don't have to check for a previous value because there is a clear spilt between field_ref and @properties
					$taxonomy_ref->{$tagid}{$prop}{$lc} = $property_value;
```

Reply:
In fact technically there could be a previous value, if the request is like fields=vegan:en&inherited_properties=vegan:en
But that would result in overriding the previous value with exactly the same value. (in fact the query would be identical to removing fields=vegan:en)